cli/command/image/build.go

```diff
@@ -94,6 +94,15 @@ func NewBuildCommand(dockerCli *command.DockerCli) *cobra.Command {
 		Args:  cli.ExactArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
 			options.context = args[0]
+
+			flags := cmd.Flags()
+			if !flags.Changed("stream") && isSessionSupported(dockerCli) {
+				// if stream is not set try the default from config file
+				if v := dockerCli.ConfigFile().BuildStreamContext; v != nil {
+					options.stream = *v
+				}
+			}
```

Comment:
Could we move this to a new function ?

```go
func setDefaultsFromConfig(dockerCli command.Cli, flags *pflag.FlatSet, options buildOptions) bool {
    ...
}
```

That will make it easy to unit test.